src/main/java/seedu/address/logic/commands/EditLogCommand.java

```diff
@@ -0,0 +1,257 @@
+            + "of an existing friend in Amigos. "
+            + "Parameters: "
+            + "INDEX ? " + PREFIX_NAME + "NAME "
+            + PREFIX_LOG_INDEX + "LOG_INDEX ["
+            + PREFIX_NEW_TITLE + "NEW_TITLE] ["
+            + PREFIX_NEW_DESCRIPTION + "NEW DESCRIPTION]\n"
+            + "Note that at least one of title and description must be provided.\n"
+            + "Example: " + COMMAND_WORD + " "
+            + "1 "
+            + PREFIX_LOG_INDEX + "1 "
+            + PREFIX_NEW_DESCRIPTION + "Likes apples";
+
+    public static final String MESSAGE_EDIT_LOG_SUCCESS = "Log successfully edited!";
+    public static final String MESSAGE_LOG_NOT_FOUND = "The specified log does not exist!";
+    public static final String MESSAGE_DUPLICATE_LOG = "This log already exists for this friend.";
+
+    // data fields
+    private final Index index;
+    private final Index logIndex;
+    private final FriendName nameToEditLog;
+    private final EditLogDescriptor editLogDescriptor;
+    private final boolean byName;
+
+    /**
+     * Creates an EditLogCommand to add the specified {@code Log} to the
+     * specified {@code Person}.
+     */
+    public EditLogCommand(Index index, Index logIndex, EditLogDescriptor editLogDescriptor) {
+        requireAllNonNull(index, logIndex, editLogDescriptor);
+        this.index = index;
+        this.nameToEditLog = null;
+        this.logIndex = logIndex;
+        this.editLogDescriptor = editLogDescriptor;
+        this.byName = false;
+    }
+
+    /**
+     * Creates an EditLogCommand to add the specified {@code Log} to the specified
+     * {@code Person}.
+     */
+    public EditLogCommand(FriendName name, Index logIndex, EditLogDescriptor editLogDescriptor) {
+        requireAllNonNull(name, logIndex, editLogDescriptor);
+        this.nameToEditLog = name;
+        this.index = null;
+        this.logIndex = logIndex;
+        this.editLogDescriptor = editLogDescriptor;
+        this.byName = true;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        Person personToEdit;
+
+        if (this.byName) {
+            personToEdit = EditLogCommand.getPersonByName(model, this.nameToEditLog);
+        } else {
+            personToEdit = EditLogCommand.getPersonByFilteredIndex(model, this.index);
+        }
+
+        // create person with edited logs
+        Person addedLogPerson = createEditedLogPerson(personToEdit, this.logIndex, this.editLogDescriptor);
+
+        // add to address book
+        model.setPerson(personToEdit, addedLogPerson);
+        return new CommandResult(MESSAGE_EDIT_LOG_SUCCESS);
+    }
+
+    private static Person createEditedLogPerson(
+            Person personToEdit, Index logIndex, EditLogDescriptor editLogDescriptor)
+            throws CommandException {
+        requireAllNonNull(personToEdit, editLogDescriptor);
```

Comment:
Should `logIndex` also be present in the `requireAllNonNull` condition here?

Reply:
Yes. Will change, thanks for catching.